students/pkoleyni/lesson04/mailroom_2.py

```diff
@@ -0,0 +1,180 @@
+    check_add_name(usr_name_input)
+    usr_donation_input = input('Enter a donation amount > ')
+    add_donation(usr_name_input, int(usr_donation_input))
+    thank_you_message(usr_name_input, usr_donation_input)
+
+
+def thank_you_message(name, donation):
+    """
+    :param name: name of a donor to send thank you letter
+    :param donation: Amount of donation
+    :return:
+    """
+    print()
+    print("Thank you letter for {}".format(name))
+    print('=' * 50)
+    print("Dear {}".format(name))
+    print("We received your {} donation".format(donation))
+    print("Thank you for your donation\n")
+
+
+def sort_key(key):
+    """
+
+    :param key: Key defines the item in the list to be used for sorting
+    :return:
+    """
+    return key[1]
+
+
+def report():
+    """
+
+    :return: will call other function to get the data and print the report
+    """
+
+    report_head = ("{:<20}{:<20}{:<20}{:<20}".format('Donor Name', '| Total Given', '| Num Gifts ', '| Average Gift'))
+    print(report_head)
+    print('-' * (len(report_head)))
+    for donor in list_donors():
+        # sorted_report_list = sorted(report_list, key = sort_key, reverse = True)
+        total_given = sum(donation_list(donor))
+        num_of_gifts = len(donation_list(donor))
+        average_gifts = total_given / num_of_gifts
+        print('{:<21}{:<21}{:<21}{:<21.2f}'.format(donor, total_given, num_of_gifts, average_gifts))
+    print(donor_list_dic)
+
+
+def send_letter_to_all():
+    write_letter_to_file('pooria koleyni')
+
+
+def file_name(name):
+    """
+
+    :param name: name of a donor
+    :return: string that will be be used as file_name for thank you letter file
+    """
+    d = datetime.datetime.today()
+    return name + '_{}_{}_{}'.format(d.year, d.month, d.day) + '.txt'
+
+
+def write_letter_to_file(name):
+    last_donation = donation_list(name)[-1]
+    line1 = 'Dear {},\n'.format(name)
+    line2 = '\tThank you for your very kind donation of ${:.2f}.\n'.format(last_donation)
+    line3 = '\tIt will be put to very good use.\n'
+    line4 = '\tSincerely,\n\t     -The Team'
+
+    file = file_name(name)
+    with open(file, 'w') as f:
+        f.write(line1.format(name))
+        f.write(line2.format(name))
+        f.write(line3.format(name))
+        f.write(line4.format(name))
+        f.close()
+    print('Letter for "{}" is saved to "{}"'.format(name, file))
+
+
+def write_letter_for_everyone():
+    for donor in list_donors():
+        write_letter_to_file(donor)
+
+
+def quit():
+    print("Quitting the menu")
+    return 'exit menu'
+
+
+def menu_selection():
+    main_prompt = """
+    Select from the following options:\n
+    1. Send a thank you
+    2. Create a report
+    3. Send_letter_to_all
+    4. Quit
+
+    """
+    main_menu_dic = {'1': thank_you,
+                     '2': report,
+                     '3': write_letter_for_everyone,
+                     '4': quit
```

Comment:
map this to `sys.exit()` so you don't have to check return value of your function below